DTXManiaプロジェクト/コード/スコア、曲/CDTX.cs

```diff
@@ -4590,9 +4472,14 @@ private void t入力_行解析ヘッダ( string InputText )
                 }
             }
             #endregion
-            else if( strCommandName.Equals( "SONGVOL" ) )
+            else if( strCommandName.Equals( "SONGVOL" ) && !string.IsNullOrEmpty( strCommandParam ) )
             {
-                //tbSongVol.Text = strCommandParam;
+                this.SongVol = Convert.ToInt32( strCommandParam ).Clamp( CSound.MinimumSongVol, CSound.MaximumSongVol );
```

Comment:
Here's where we bring SONGVOL back online, supported elsewhere by the various updates in sound level control that eventually flow through to CSound and then down into the audio playback library.

Reply:
私はSONGVOL and SEVOLの最大値は100ではないと思います。
常識的には100である必要がありますが、ごく一部の.tjaファイルには101以上の値が割り当てられます。
Taikojiroではそのとき、音源を増幅して再生します。
個人的には100以降の値はサポートする必要がないので、このままで良いです。
.tjaファイル内のSONGVOL and SEVOLを無視する設定も、できればあったほうが良いです。

I think that the maximum value of SONGVOL and SEVOL is not 100.
It is common sense to be 100, but only a small number of .tja files are assigned a value of 101 or more.
At that time, Taikojiro amplifies the sound source and plays it.
Personally, since you do not need to support values ​​after 100, you can stay as it is.
It is better to have settings to ignore SONGVOL and SEVOL in the .tja file, if possible.

Reply:
Your comment is perfect. :)

I capped them at 100 for now because I didn't want to risk any clipping issues, BUT... In my next round of work I should be able to remove that cap because the mixing approach is going to change from linear mixing with multiplication of percentages to logarithmic mixing with addition of relative loudness, and for that I have to change to a different numerical range and will need to allow for values above full scale.

The other reason your comment is perfect is because I'll also be adding config support to control whether or not and at what priority SONGVOL is even applied relative to the loudness metadata support I am adding.

As for SEVOL, I personally don't plan to support it. I can't really see a reason to once the loudness metadata support is in place, because really quiet song files won't be really quiet any more, and so there would be no reason to reduce the SEVOL for a given tja. ;)